src/material-experimental/mdc-core/option/_optgroup-theme.scss

```diff
@@ -0,0 +1,46 @@
+@import '@material/list/mixins.import';
+@import '@material/list/variables.import';
+@import '@material/theme/functions.import';
+@import '../../mdc-helpers/mdc-helpers';
+@import '../../../material/core/theming/check-duplicate-styles';
+
+@mixin mat-mdc-optgroup-color($config-or-theme) {
+  $config: mat-get-color-config($config-or-theme);
+
+  @include mat-using-mdc-theme($config) {
+    .mat-mdc-optgroup-label {
+      // Since this will usually we rendered in an overlay,
+      // we have explicitly set the default color.
+      @include mdc-theme-prop(color, text-primary-on-background);
+      @include mdc-list-item-disabled-text-color($mdc-list-text-disabled-color,
+        $query: $mat-theme-styles-query);
+    }
+  }
+}
+
+@mixin mat-mdc-optgroup-typography($config-or-theme) {
+  $config: mat-get-typography-config($config-or-theme);
+}
+
+@mixin _mat-optgroup-density($config-or-theme) {
```

Comment:
We don't prefix the density mixins in `@angular/material-experimental`, so we should remove that here, and also add the missing `mat-mdc` prefix.